### ISSUE TITLE
`gevent` integration does not work with `gevent.pool.Group.map`

### ISSUE THREAD

**Issue by wackywendell** [DataDog/dd-trace-py#275]
The context is not properly passed from parent greenlet to child greenlet when using `gevent.pool.Group().map`. Spawning greenlets directly works:

```
pool = gevent.pool.Group()

# Context passes correctly
greenlets = [pool.spawn(myfunc, arg) for arg in args]
results = [g.get() for g in greenlets]

# Context does not pass correctly
results = pool.map(myfunc, args)
```

After some debugging, I tracked it to [this point in `ddtrace`](https://github.com/DataDog/dd-trace-py/blob/d2b1ef00597709876f3f479a31bbaf8e65663632/ddtrace/contrib/gevent/provider.py#L25-L26). 

Basically, before spawning a `TracedGreenlet`, it asks `gevent` for the current greenlet, and takes the context attached to that greenlet and attaches it to the new greenlet. However, when using `pool.map`, the "current" greenlet is not a greenlet; its an instance of [`IMap`](https://github.com/gevent/gevent/blob/f3acb176d0f0f1ac797b50e44a5e03726f687c53/src/gevent/pool.py#L181), which has not inherited the context.

**Comment by palazzem** [comment]
Thank you @wackywendell for spotting that. Will update our instrumentation to handle the `pool.map` too!

**Comment by brettlangdon** [comment]
This has been resolved by #600
